fbw-a32nx/src/systems/fmgc/src/flightplanning/FlightPlanManager.ts

```diff
@@ -1032,7 +1032,7 @@ export class FlightPlanManager {
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     public truncateWaypoints(index: number, thenSetActive = false, callback = () => { }): void {
         const fp = this._flightPlans[this._currentFlightPlanIndex];
-        for (let i = fp.length; i >= index; i--) {
+        for (let i = fp.length; i > index; i--) {
```

Comment:
Shouldn't it start at `fp.length - 1` (i.e. the last index in the plan), and the original condition is then correct?

Reply:
That would also work (but is one extra op ;-)).  `index` doesn't change so the loop is calling `removeWaypoint(index)` n times.  The first time in the loop the waypoint[index] is removed, that matches the doc. Remaining waypoints "shift left" so each time after that the succeeding waypoint is removed.  Just the calculation of n was wrong.